File: solutions/7/sol.py
def get_prime_numbers(n, arr):
  """
  Get all prime numbers from 0 to n*2 (inclusive).

  Parameters:
  n (int): The upper limit multiplier for finding primes (range is 0 to n*2).
  arr (list[int]): All prime numbers from 0 to n (inclusive).

  Returns:
  list[int]: A list of prime numbers between 0 and n*2 (inclusive).
  """
  is_primes = [True] * (n * 2 + 1)
  return is_primes

prime_numbers = get_prime_numbers(2, [2])
print(prime_numbers)

File: solutions/7/sol.js
/**
 * Get all prime numbers from 0 to n*2
 * @param {*} n 
 * @param {*} headPrimeNumbers all prime numbers from 0 to n
 */
const getPrimeNumbers = (n, headPrimeNumbers) => {
  const length = 2 * n + 1;
  const isPrimes = new Array(length).fill(true);
  for (const primeNumber of headPrimeNumbers) {
    let num = primeNumber;
    while (num < length) {
      isPrimes[num] = false;
      num += primeNumber;
    }
  }
  const tailPrimeNumbers = [];
  for (let num = n + 1; num < length; num++) {
    if (isPrimes[num]) {
      tailPrimeNumbers.push(num);
    }
  }
  return [...headPrimeNumbers, ...tailPrimeNumbers];
}
console.log(getPrimeNumbers(2, [2]));
console.log(getPrimeNumbers(4, [2, 3]));
console.log(getPrimeNumbers(8, [2, 3, 5, 7]));
console.log(getPrimeNumbers(16, [2, 3, 5, 7, 11, 13]));